tests/Unit/ApparentHorizons/TestHorizonManagerHelper_ElementComponent.hpp

```diff
@@ -0,0 +1,125 @@
+
+#include <tuple>
+#include <utility>
+
+#include "ApparentHorizons/HorizonManagerComponent.hpp"
+#include "ApparentHorizons/HorizonManagerComponentActions.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/CreateInitialElement.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/DomainCreators/DomainCreator.hpp"
+#include "Domain/Element.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/ElementIndex.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/InitialElementIds.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Parallel/Algorithm.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Info.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Printf.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/KerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/GrTags.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/ApparentHorizons/TestHorizonManagerHelper_ElementActions.hpp"
+
+#include <iostream>
+
+template <class Metavariables>
+struct DgElementArray {
+  using chare_type = Parallel::Algorithms::Array;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using metavariables = Metavariables;
+  using action_list = tmpl::list<>;
+  using array_index = ElementIndex<3>;
+  using initial_databox = db::compute_databox_type<
+      typename Actions::DgElementArray::InitializeElement::return_tag_list>;
+  using options = tmpl::list<typename Metavariables::domain_creator_tag>;
+  static void initialize(
+      Parallel::CProxy_ConstGlobalCache<Metavariables>& global_cache,
+      std::unique_ptr<DomainCreator<3, Frame::Inertial>>
+          domain_creator) noexcept;
+  static void execute_next_phase(
+      const typename Metavariables::Phase next_phase,
+      const Parallel::CProxy_ConstGlobalCache<Metavariables>& global_cache) {
+    if (next_phase == Metavariables::Phase::InitialCommunication) {
+      auto& dg_element_array = Parallel::get_parallel_component<DgElementArray>(
+          *(global_cache.ckLocalBranch()));
+
+      Parallel::simple_action<Actions::DgElementArray::SendNumElements<
+          HorizonManagerComponent<Metavariables>>>(dg_element_array);
+    }
+    if (next_phase == Metavariables::Phase::CheckAnswer) {
+      auto& my_proxy = Parallel::get_parallel_component<
+          HorizonManagerComponent<Metavariables>>(
+          *(global_cache.ckLocalBranch()));
+      Parallel::simple_action<Actions::HorizonManager::PrintNumElements>(
+          my_proxy);
```

Comment:
I suggest you add an action that does a few `SPECTRE_PARALLEL_REQUIRE`s to test results, like in `Test_AlgorithmReduction.cpp` for instance. In particular also in another phase after the horizon finding.